### PR TITLE
Paywalls Tester: Universal links (#1 in queue)

### PR DESCRIPTION
**Pull request by jamesrb1**
This adds support for universal links.

Format was
app.revenuecat.com/paywallstester/?pw=ofrng71bdfc2037

but should soon be
app.revenuecat.com/pwt/?pw=ofrng71bdfc2037

QR code for the second one:

image

### CODE REVIEW

File: Tests/TestingApps/PaywallsTester/PaywallsTester/Configuration.swift [hunk 1]
```diff
@@ -33,7 +33,7 @@ final class Configuration: ObservableObject {
     #warning("Configure API key if you want to test paywalls from your dashboard")
     // Note: you can leave this empty to use the production server, or point to your own instance.
     private static let proxyURL = ""
-    private static let apiKey = ""
+    private static let apiKey = "appl_TFIZqBlauHeMhnqfNlWWFBjZwkf"
```
Comment (aboedo) [inline Tests/TestingApps/PaywallsTester/PaywallsTester/Configuration.swift:36]: ❗

File: Tests/TestingApps/PaywallsTester/PaywallsTester/Data/OfferingsPaywallsViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+        showPaywallForID(id)
+    }
+
+    private static var logger = Logging.shared.logger(category: "Paywalls Tester")
+
+}
+
+// Private helpers
+extension OfferingsPaywallsViewModel {
```
Comment (aboedo) [inline Tests/TestingApps/PaywallsTester/PaywallsTester/Data/OfferingsPaywallsViewModel.swift:79]: might as well make this a private extension and then the modifier isn't needed anymore

File: Tests/TestingApps/PaywallsTester/PaywallsTester/SimpleApp.swift [hunk 3]
```diff
@@ -2,22 +2,68 @@
...
 //  SimpleApp
 //
-//  Created by Nacho Soto on 5/30/23.
+//  Created by James Borthwick on 4/25/24.
 //
 
 import SwiftUI
 
 @main
 struct SimpleApp: App {
```
Comment (aboedo) [inline Tests/TestingApps/PaywallsTester/PaywallsTester/SimpleApp.swift:11]: this name is suuuper outdated, we set up the name "SimpleApp" when we were thinking about having a simple app that didn't have anything but a paywall on it for testing. Might as well rename it to something more appropriate

File: Tests/TestingApps/PaywallsTester/PaywallsTester/SimpleApp.swift [hunk 4]
```diff
@@ -2,22 +2,68 @@
...
+                }
         }
+        .environment(application)
+    }
+
+}
+
+// MARK: - Universal Links
+extension SimpleApp {
+    func processURL(_ url: URL) {
```
Comment (aboedo) [inline Tests/TestingApps/PaywallsTester/PaywallsTester/SimpleApp.swift:50]: nitpick - single line break at start and end of declaration

**Review by aboedo — APPROVED** [review]

File: Tests/TestingApps/PaywallsTester/PaywallsTester/Configuration.swift (hunk 1 above)
Comment (jamesrb1) [reply Tests/TestingApps/PaywallsTester/PaywallsTester/Configuration.swift:36]: How should I include this for the TF build?

File: Tests/TestingApps/PaywallsTester/PaywallsTester/Data/OfferingsPaywallsViewModel.swift (hunk 2 above)
Comment (jamesrb1) [reply Tests/TestingApps/PaywallsTester/PaywallsTester/Data/OfferingsPaywallsViewModel.swift]: Done!

File: Tests/TestingApps/PaywallsTester/PaywallsTester/SimpleApp.swift (hunk 3 above)
Comment (jamesrb1) [reply Tests/TestingApps/PaywallsTester/PaywallsTester/SimpleApp.swift]: I used `PaywallsPreview`, which I think is a better description than `PaywallsTester`.

File: Tests/TestingApps/PaywallsTester/PaywallsTester/Configuration.swift (hunk 1 above)
Comment (aboedo) [reply Tests/TestingApps/PaywallsTester/PaywallsTester/Configuration.swift:36]: CircleCI env variable + variable replacement in a script, you can see how we do it for the backend integration tests app

**Pull request merged**